docs/features/network-security-controls/dns-name-resolution.md

```diff
@@ -0,0 +1,84 @@
+# DNS name resolover
+
+## Introduction
+
+The DNS name resolver feature allows a cluster administrator to enable integration of
+EgressFirewall with CoreDNS. Once enabled the EgressFirewall DNS rules will be resolved
+using CoreDNS and the underlying address set will be updated accordingly. Additionally,
+if any other pod does the DNS resolution of DNS names used in the EgressFirewall DNS
+rules, then the underlying address sets will also be updated if there is any change in
+IP addresses.
+
+The DNS name resolver feature also adds the support of using wildcard DNS names in
+EgressFirewall DNS name rules. The wildcard (`*`) will match only one label (subdomain).
+For example, `*.example.com` will match `sub1.example.com` and will not match
+`sub2.sub1.example.com`.
+
+The [`DNSNameResolver`](https://github.com/openshift/api/tree/ef21ee7c3d0590ac431e81059172615e2addbbe3/network/v1alpha1/zz_generated.crd-manifests)
+CRD will be used by ovnk to get the latest IP addresses corresponding to a DNS name when
+the feature is enabled. However, the cluster administrator does not need to interact with
+the CR. 
+
+## Configuring DNS name resolover in a cluster
+
+The DNS name resolver should be configured in the following components for it to work
+in a cluster:
+
+### [DNSNameResolver operator](https://github.com/openshift/coredns-ocp-dnsnameresolver/tree/main/operator)
```

Comment:
I think we will automate this in our kind.sh script, so maybe we can omit this section for now